activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -64,12 +64,18 @@ def initialize(constructor = {})
       end
     end
 
-    def default(key = nil)
-      if key.is_a?(Symbol) && include?(key = key.to_s)
-        self[key]
-      else
-        super
-      end
+    def default(*args)
```

Comment:
@georgeclaghorn `Hash#default` is wrongly documented as `nil` is a valid argument. 

See the [Rubinius implementation](https://github.com/rubinius/rubinius/blob/master/kernel/common/hash.rb#L272-L278).